addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -186,14 +191,129 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @see this.selectClass for params
      */
-    style: function (previewMode, widgetValue, params) {
-        this._rpc({
-            route: '/shop/change_styles',
-            params: {
-                'id': this.productTemplateID,
-                'style_id': params.possibleValues[params.possibleValues.length - 1],
-            },
-        });
+    async setRibbon(previewMode, widgetValue, params) {
+        this.$target[0].dataset.ribbonId = widgetValue;
+        if (this.ribbons[widgetValue]) {
+            await this._rpc({
+                route: '/shop/set_ribbon',
+                params: {
+                    'product_tmpl_id': this.productTemplateID,
+                    'ribbon_id': widgetValue || false,
+                },
+            });
```

Comment:
Not sure what you mean here.

Reply:
Add a ribbon on a product, hit discard, the ribbon is still saved. The ribbon was saved directly so not possible to discard.

Reply:
Right! Previously, it would reload the page as well. There is a task in india for something similar: task-1960780

I'll think about it, see if it can be done in a more generic way that can be used by other options that change config on the server.

Reply:
Yeah I know I have to take a look at it again, but it is not similar as in their case it is not possible to not reload. The task changed to only make the reload more pretty (odoo modal instead of ugly browser one).

As yours is adding or removing a generic website item (a ribbon) (so not reloading a complex grid with internal t-field items), it is more possible :)